From 51160640418cb29a3d9423617df753bd1b23109a Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_github@freenet.de>
Date: Wed, 9 Mar 2011 00:00:00 +0100
Subject: [PATCH 01/14] Nit picked in 2011.

---
 miles_span.w | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/miles_span.w b/miles_span.w
index d576770..6a5f9b7 100644
--- a/miles_span.w
+++ b/miles_span.w
@@ -684,7 +684,7 @@ Vertex *del_heap()
   register Vertex *u; /* vertex being sifted down */
   register unsigned long k; /* hole in the heap */
   register unsigned long j; /* child of that hole */
-  register long d; /* |u->dist|, the vertex of the vertex being sifted */
+  register long d; /* |u->dist|, the key of the vertex being sifted */
   if (hsize==0) return NULL;
   o,v=heap_elt(1);
   o,u=heap_elt(hsize--);
-- 
2.19.2

