From cc30b080be885ef69c80d0e22008871ebe1081ab Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_github@freenet.de>
Date: Thu, 6 Aug 2015 00:00:00 +0200
Subject: [PATCH 04/14] Nit picked in 2015.

---
 gb_words.w | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gb_words.w b/gb_words.w
index 994055f..94a049a 100644
--- a/gb_words.w
+++ b/gb_words.w
@@ -396,7 +396,7 @@ done:gb_free(node_blocks);
 @ The only slightly unusual data structure needed is a set of five hash tables,
 one for each of the strings of four letters obtained by suppressing
 a single letter of a five-letter word. For example, a word like `\.{words}'
-will lead to entries for `\.{\ ords}', `\.{w\ rds}, `\.{wo\ ds}', `\.{wor\ s}',
+will lead to entries for `\.{\ ords}', `\.{w\ rds}', `\.{wo\ ds}', `\.{wor\ s}',
 and `\.{word\ }', one in each of the hash tables.
 
 @d hash_prime 6997 /* a prime number larger than the total number of words */
-- 
2.20.1

