From 51130162a84827135184a9075de63ce279c6aa05 Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Sat, 20 Jan 2024 11:53:20 +0100
Subject: [PATCH 25/25] Separate do..while loops.

---
 PROTOTYPES/gb_dijk.ch | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/PROTOTYPES/gb_dijk.ch b/PROTOTYPES/gb_dijk.ch
index 911e4ea..c752be8 100644
--- a/PROTOTYPES/gb_dijk.ch
+++ b/PROTOTYPES/gb_dijk.ch
@@ -113,6 +113,12 @@ void print_dijkstra_result(
   } while (t);
 @z
 
+@x l.275
+  t=p;
+@y
+  @/t=p;@/
+@z
+
 @x l.276
   do@+{ /* pop an item from |t| to |p| */
 @y
-- 
2.43.0

