From 7011fe4b87d99ebc73e8008669610b5f74ca2738 Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Tue, 25 Jun 2024 09:13:42 +0200
Subject: [PATCH 30/32] Issue #13: Prototypical example code.

---
 PROTOTYPES/gb_plane.ch | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/PROTOTYPES/gb_plane.ch b/PROTOTYPES/gb_plane.ch
index 697bd99..52763ae 100644
--- a/PROTOTYPES/gb_plane.ch
+++ b/PROTOTYPES/gb_plane.ch
@@ -78,6 +78,15 @@ for (k=0,v=new_graph->vertices; k<(long)n; k++,v++) {
 @ (This section remains empty for historic reasons.)
 @z
 
+@x [9] l.215
+|void ins_finite(u,v)|\cr
+\qquad|Vertex *u,*v;|\cr
+|{@+if (u&&v)@+gb_new_edge(u,v,1L);@+}|\cr}}$$
+@y
+|@,void ins_finite(Vertex *u, Vertex *v)|\cr
+|{@+if (u&&v)@+gb_new_edge(u,v,1L);@+}|\cr}}$$
+@z
+
 @x [9] l.226
 void delaunay(g,f)
   Graph *g; /* vertices in the plane */
-- 
2.45.2

