From d75ad641fb1675250b7e002d6ea24eebf3984590 Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Wed, 26 Jun 2024 09:50:10 +0200
Subject: [PATCH 31/33] Fix line break with 'cweave -f'.

---
 PROTOTYPES/gb_gates.ch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/PROTOTYPES/gb_gates.ch b/PROTOTYPES/gb_gates.ch
index 60347bc..49eea27 100644
--- a/PROTOTYPES/gb_gates.ch
+++ b/PROTOTYPES/gb_gates.ch
@@ -90,7 +90,7 @@ long gate_eval(
 @x [3] l.144
     @t\4\4@>@<Compute the value |t| of a classical logic gate@>;
 @y
-    @t\4@>@<Compute the value |t| of a classical logic gate@>@;
+  @/@t\4@>@<Compute the value |t| of a classical logic gate@>@;
 @z
 
 @x [7] l.184
-- 
2.45.2

