From 4fe1ad6a6c921b0185369f56d482ed1ebc9af77c Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Wed, 3 Jul 2024 10:16:57 +0200
Subject: [PATCH 33/39] List changed sections in Editor's Note.

---
 boilerplate.w | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/boilerplate.w b/boilerplate.w
index 91d3aef..a6c3256 100644
--- a/boilerplate.w
+++ b/boilerplate.w
@@ -15,7 +15,8 @@ incarnation.
 \smallskip\noindent Consult the online project
 \pdfURL{{\ninett https://github.com/ascherer/sgb}}{https://github.com/ascherer/sgb}
 for context and rationale.
-\vskip1in
+\readchanges % The following sections were changed...
+\vskip 0pt plus 1filll
     \noindent\copyright\ 1993 Stanford University
     \bigskip\noindent
     This file may be freely copied and distributed, provided that
@@ -45,3 +46,20 @@ for context and rationale.
     Important: Before reading {\sc\title}, please read
     or at least skim the programs for {\sc#1} and {\sc#2}.\bigskip
     \let\startsection=\stsec\stsec}}
+
+\def\changedsections{\jobname.chs}
+\def\readchanges{\input \changedsections}
+\newread\testread \newwrite\writechanges
+\openin\testread=\changedsections\relax
+\ifeof\testread % First run
+  \let\readchanges=\relax
+  \let\CH=\ch
+  \def\ch#1.{\begingroup
+    \let\\=\BS \let\*=\let \let\ch=\relax
+    \immediate\openout\writechanges=\changedsections
+    \immediate\write\writechanges{\par\noindent\ch#1.}
+    \immediate\closeout\writechanges
+    \let\ch=\CH \ch#1.
+    \endgroup}
+\else\closein\testread % Second run
+\fi
-- 
2.51.0

