From d2a6747c7d31166cc8f065f333127af64ed4e2d0 Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_github@freenet.de>
Date: Tue, 19 May 2026 19:28:59 +0200
Subject: [PATCH 67/73] Compile girth with C++23.

---
 PROTOTYPES/girth.ch | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/PROTOTYPES/girth.ch b/PROTOTYPES/girth.ch
index 83b6d38..16adc3c 100644
--- a/PROTOTYPES/girth.ch
+++ b/PROTOTYPES/girth.ch
@@ -10,3 +10,19 @@ main()
 @y
 int main(void)
 @z
+
+@x [12] l.303
+  register Vertex *v; /* current vertex in list at distance $k-1$ */
+  register Vertex *u; /* head of list for distance $k$ */
+@y
+  Vertex *v; /* current vertex in list at distance $k-1$ */
+  Vertex *u; /* head of list for distance $k$ */
+@z
+
+@x [13] l.321
+{@+register Arc *a;
+  for (a=v->arcs;a;a=a->next) {@+register Vertex *w;
+@y
+{@+Arc *a;
+  for (a=v->arcs;a;a=a->next) {@+Vertex *w;
+@z
-- 
2.54.0

